decode.rb

```diff
@@ -0,0 +1,46 @@
+CODES = {
+  A: '.-',
+  B: '-...',
+  C: '-.-.',
+  D: '-..',
+  E: '.',
+  F: '..-.',
+  G: '--.',
+  H: '....',
+  I: '..',
+  J: '.---',
+  K: '-.-',
+  L: '.-..',
+  M: '--',
+  N: '-.',
+  O: '---',
+  P: '.--.',
+  Q: '--.-',
+  R: '.-.',
+  S: '...',
+  T: '-',
+  U: '..-',
+  V: '...-',
+  W: '.--',
+  X: '-..-',
+  Y: '-.--',
+  Z: '--..'
+}.freeze
+def
+ decode_char(input)
+  individual_input = input.split('   ')
+  message = []
+  individual_input.each do |i|
+    message.push(i)
+  end
+  create_message(message)
+end
+
+def create_message(message)
+  decoded_message = message.map do |word|
+    word.split.map { |code| CODES.key(code) }.join
+  end
+  puts decoded_message.join(' ')
+end
+
```

Comment:
- [ ] According to the project requirements, you should have created:
-  `#decode_char` method:
   * it translates Morse code symbols into letters
- `#decode_word` method:
  * it splits input word by a single space 
  * it reuses the `#decode_char` method to translate a single char symbol into a letter (DRY rule)
  * it combines translated chars into a single word to be returned as the final output
- and finally `#decode` method and works as expected:
   * it splits input sentences by 3 spaces (word delimiter in Morse code)
  * it reuses the `#decode_word` method to translate a single encoded-word into a word (DRY rule)
  * it combines translated words into a whole sentence to be returned as the final output.
-  let's address this 👍🏻